lib/model.js

```diff
@@ -1395,6 +1395,15 @@ Model.createCollection = async function createCollection(options) {
     }
   }
 
+  const clusteredIndex = this &&
+    this.schema &&
+    this.schema.options &&
+    this.schema.options.clusteredIndex;
+  if (clusteredIndex != null) {
+    options = Object.assign({ clusteredIndex }, options);
+    options.clusteredIndex.unique = true;
```

Comment:
Best to clone the `clusteredIndex` value before modifying it, `options.clusteredIndex = { ...clusteredIndex, unique: true }`

Reply:
no problem!